apps/src/templates/rubrics/teacherRubricRedux.ts

```diff
@@ -0,0 +1,88 @@
+import {PayloadAction, createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+
+interface TeacherEvaluation {
+  feedback: string;
+  id: number;
+  learning_goal_id: number;
+  understanding: number;
+}
+
+interface UserEvalData {
+  eval: TeacherEvaluation[];
+  user_family_name?: string;
+  user_id: number;
+  user_name: string;
+}
+
+type AllTeacherEvaluationData = UserEvalData[];
+
+interface HasTeacherFeedbackMap {
+  [key: number]: boolean;
+}
+
+interface TeacherRubricState {
+  allTeacherEvaluationData: AllTeacherEvaluationData;
+  hasTeacherFeedbackMap: HasTeacherFeedbackMap;
+}
+
+const initialState: TeacherRubricState = {
+  allTeacherEvaluationData: [],
+  hasTeacherFeedbackMap: {},
+};
+
+const teacherRubricReduxSlice = createSlice({
+  name: 'teacherRubric',
+  initialState,
+  reducers: {
```

Comment:
Adding this would let you avoid exposing the full `setHasTeacherFeedbackMap` to external components, and could be called directly from the `RubricSubmitFooter` that handles this interaction.

```suggestion
  reducers: {
    registerTeacherFeedbackForUser(state, action: PayloadAction<number>) {
      state.hasTeacherFeedbackMap[action.payload] = true;
    },
```

Reply:
done as `setUserHasTeacherFeedback`